source/gridfs/gridfs-spec.md

```diff
@@ -0,0 +1,1021 @@
+for the given filename, sorted by uploadDate (either ascending or descending, depending on the revision requested) and
+skipping the appropriate number of documents. For negative revision numbers, the sort is descending and the number of
+documents to skip equals (-revision - 1). For non-negative revision numbers, the sort is ascending and the number of
+documents to skip equals the revision number.
+
+If a networking error or server error occurs, drivers MUST raise an error.
+
+### Partial File Retrieval
+
+In the case of open_download_stream, drivers SHOULD support partial file retrieval by allowing the application to read
+only part of the stream. If a driver does support reading only part of the stream, it MUST do so using the standard
+stream methods of its language for seeking to a position in a stream and reading the desired amount of data from that
+position. This is the preferred method of supporting partial file retrieval.
+
+In the case of download_to_stream, drivers are not required to support partial file retrieval. If they choose to do so,
+drivers can support this operation by adding "start" and "end" to their supported options for download_to_stream. These
+values represent non-negative byte offsets from the beginning of the file. When "start" and "end" are specified, drivers
+return the bytes of the file in `[start, end)`. If "start" and "end" are equal no data is returned.
+
+If either "start" or "end" is invalid, drivers MUST raise an error. These values are considered invalid if they are
+negative, greater than the file length, or if "start" is greater than "end".
+
+When performing partial reads, drivers SHOULD use the file"s "chunkSize" to calculate which chunks contain the desired
+section and avoid reading unneeded documents from the "chunks" collection.
+
+### Renaming stored files
+
+```javascript
+class GridFSBucket {
+
+  /**
+   * Renames the stored file with the specified @id.
+   */
+  void rename(TFileId id, string new_filename);
+
+}
+```
+
+Sets the filename field in the stored file"s files collection document to the new filename.
+
+**Implementation details:**
+
+Drivers construct and execute an update_one command on the files collection using `{ _id: @id }` as the filter and
+`{ $set : { filename : "new_filename" } }` as the update parameter.
+
+To rename multiple revisions of the same filename, users must retrieve the full list of files collection documents for a
+given filename and execute "rename" on each corresponding `_id`.
+
+If there is no file with the given id, drivers MUST raise an error.
+
+### Dropping an entire GridFS bucket
+
+```javascript
+class GridFSBucket {
+
+  /**
+   * Drops the files and chunks collections associated with
+   * this bucket.
+   */
+  void drop();
+
+}
+```
+
+This method drops the files and chunks collections associated with this GridFS bucket.
+
+Drivers should drop the files collection first, and then the chunks collection.
+
+## Test Plan
+
+TBD
```

Comment:
This can either be removed or replaced with a link to `tests/README.md`. I realize this goes back to the RST file, but now is a good time to address it.

Removing is probably easier.

Reply:
Removed